src/focus.service.ts

```diff
@@ -509,6 +509,17 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
+    let visible = true;
+    for (let parent = el; parent !== this.root; parent = parent.parentElement) {
+      if (parent.classList.contains('hidden')) {
+        visible = false;
+        break;
+      }
+    }
+
+    if (!visible) {
+      return false;
+    }
```

Comment:
If the element is invisible, we add it to a list of exclusions and attempt to find a new element, excluding the one we just pulled. This can be implemented recursively or iteratively without too much work.